cpp/vm/evmzero/interpreter.cc

```diff
@@ -1487,6 +1489,22 @@ struct CreateImpl {
     if (gas -= mem_cost; gas < 0) [[unlikely]]
       return {.dynamic_gas_costs = initial_gas - gas};
 
+    if (ctx.revision >= EVMC_SHANGHAI) {
+      const int64_t max_code_size = 24576;
+      const int64_t max_init_code_size = 2 * max_code_size;
+      const int64_t init_code_word_cost = 2;
+
+      if (init_code_size > max_init_code_size) {
+        return {
+            .state = RunState::kErrorInitCodeSizeExceeded,
+            .dynamic_gas_costs = initial_gas - gas,
+        };
+      }
+      if (gas -= init_code_word_cost * static_cast<int64_t>(((init_code_size + 31) / 32)), gas < 0) {
```

Comment:
nit: i think it would help readability to decompose this into two steps:
```suggestion
      const int64_t init_code_costs = init_code_word_cost * static_cast<int64_t>(((init_code_size + 31) / 32));
      if (gas -= init_code_costs, gas < 0) {
```